charts/cdnsd/templates/deployment.yaml

```diff
@@ -0,0 +1,75 @@
+---
+apiVersion: apps/v1
+kind: Deployment
```

Comment:
I would use a StatefulSet, as we have optional persistence. With a Deployment, it'll create a single persistent volume and share it between Pods. This only works with ReadWriteMany volumes (like NFS/network storage) and not ReadWriteOnce volumes (like EBS/block storage). With a StatefulSet, it'll create a volume per replica, and they'll be paired.